### PR TITLE
Company contacts collection list

### PR DESCRIPTION
**Pull request by paulgain**
## Description of change

Complete rewrite the company contact collection list in React.

## Test instructions

Go to `/companies/<company-id>/contacts`

## Screenshots
Screenshot 2021-09-07 at 08 29 56

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/contacts/client/transformers.js [hunk 1]
```diff
@@ -12,53 +12,41 @@ const getTelephoneNumber = (telephone_countrycode, telephone_number) =>
...
-  company,
-  modified_on,
-  archived,
-  company_sector,
-  primary,
-  telephone_countrycode,
-  telephone_number,
-  email,
-} = {}) => {
+export const transformContactToListItem = (companyId) => (contact) => {
```
Comment (ian-leggett) [inline src/apps/contacts/client/transformers.js:15]: Really small but you could deconstruct "contact" to remove repetition below.

**Review by ian-leggett — APPROVED** [review]

**Comment by ian-leggett** [comment]
Looks really good. Left one small comment thats all. I think an old test maybe running as its looking for a class selector `c-entity-list__item` hence the failing test.

**Review by ian-leggett — COMMENTED** [review]
Looks really good, should we test the payload like we did in the other collection lists? Also I think an old test maybe running as its looking for a class selector `c-entity-list__item` hence the failing test.

File: src/apps/contacts/client/transformers.js (hunk 1 above)
Comment (paulgain) [reply src/apps/contacts/client/transformers.js:15]: Yep, it was originally deconstructed, however, I feel it's easier to read using the `contact` reference as it reduces the the function line count by 14.

File: src/client/components/FilteredCollectionList/index.jsx [hunk 2]
```diff
@@ -37,6 +37,7 @@ const FilteredCollectionList = ({
   addItemUrl,
   defaultQueryParams,
   titleRenderer = null,
+  width = 'two-thirds',
```
Comment (tomkingston) [inline src/client/components/FilteredCollectionList/index.jsx:40]: The investment opportunities page uses the collection list, but didn't need to set a width param (https://www.datahub.trade.gov.uk/investments/opportunities), so might be nice if passing width can be avoided (ie the css works out the width according to the space available), but not worth spending much time on

**Review by tomkingston — APPROVED** [review]

File: src/client/components/FilteredCollectionList/index.jsx (hunk 2 above)
Comment (paulgain) [reply src/client/components/FilteredCollectionList/index.jsx]: Good spot! I've removed the width setting altogether, initially it was set to `two-thirds` as the other third was consumed by the filters, but really there's no need to set the width whatsoever.

**Pull request merged**